connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/DataStreamReader.scala

```diff
@@ -217,6 +217,20 @@ final class DataStreamReader private[sql] (sparkSession: SparkSession) extends L
    */
   def parquet(path: String): DataFrame = format("parquet").load(path)
 
+  /**
+   * Define a Streaming DataFrame on a Table. The DataSource corresponding to the table should
+   * support streaming mode.
+   * @param tableName The name of the table
+   * @since 3.5.0
+   */
+  def table(tableName: String): DataFrame = {
+    require(tableName != null, "The table name can't be null")
+    sparkSession.newDataFrame { builder =>
+      builder.getReadBuilder.setIsStreaming(true).getNamedTableBuilder
```

Comment:
Need set options.

Reply:
Also, it will more readable to split this into more than one line.

Reply:
Sure I'll make it multiple lines. For options I thought they are already set in the option method? 
https://github.com/apache/spark/blob/ad58f4c99eb9957d8f6bf55aa8dc7f7d48692f31/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/DataStreamReader.scala#L82

Here we don't have the `extraOptions` like what did in `DataFrameReader`

Reply:
Is there a way to also test the options? Maybe `spark.readStream().option().table()`?

Reply:
On second thought I think we do need to set options. I've updated the code. I had a hard time finding an option for table though..

Reply:
You can refer to https://github.com/apache/spark/pull/40498 to how that is tested.

Reply:
Cool thank you!